aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -0,0 +1,433 @@
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct Object has key {
+        guid_creation_num: u64,
+        owner: address,
+        allow_ungated_transfer: bool,
+        transfer_events: event::EventHandle<TransferEvent>,
+    }
+
+    #[resource_group(scope = global)]
+    struct ObjectGroup { }
+
+    /// Type safe way of designate an object as at this address
+    struct ObjectId has copy, drop, store {
+        inner: address,
+    }
+
+    /// This is a one time ability given to the creator to configure the object as necessary
+    struct CreatorRef has drop {
+        self: ObjectId,
+        can_delete: bool,
+    }
+
+    /// The owner of this ability can delete an object
+    struct DeleteRef has drop, store {
+        self: ObjectId,
+    }
+
+    /// Access to this ref allows for creation of the signer to add new resources
+    struct ExtendRef has drop, store {
+        self: ObjectId,
+    }
+
+    /// Access to this ref allows for owner to transfer the resources between two addresses
+    struct TransferRef has drop, store {
+        self: ObjectId,
+    }
+
+    /// Access to this ref allows for owner to transfer the resources between two addresses
+    struct LinearTransferRef has drop {
+        self: ObjectId,
+    }
+
+    struct TransferEvent has drop, store {
+        object_id: ObjectId,
+        from: address,
+        to: address,
+    }
+
+    /// Produces an ObjectId from the given address.
+    public fun create_object_id(object_id: address): ObjectId {
+        ObjectId { inner: object_id }
+    }
+
+    /// Derives an object id from source material: sha3_256([creator address | seed | 0xFE]).
+    /// The ObjectId needs to be distinct from create_resource_address
+    public fun derive_object_id(source: &address, seed: vector<u8>): ObjectId {
+        let bytes = bcs::to_bytes(source);
+        vector::append(&mut bytes, seed);
+        vector::push_back(&mut bytes, OBJECT_ADDRESS_SCHEME);
+        ObjectId { inner: from_bcs::to_address(hash::sha3_256(bytes)) }
+    }
+
+
+    /// Returns the address of an ObjectId
+    public fun object_id_address(object_id: &ObjectId): address {
+        object_id.inner
+    }
+
+    /// Create a new named object and return the CreatorRef. Named objects can be queried globally
+    /// by knowing the user generated seed used to create them. Named objects cannot be deleted.
+    public fun create_named_object(creator: &signer, seed: vector<u8>): CreatorRef {
+        let creator_address = signer::address_of(creator);
+        let id = derive_object_id(&creator_address, seed);
+        create_object_internal(creator_address, id, false)
+    }
+
+    public fun create_object_from_account(creator: &signer): CreatorRef {
+        let guid = account::create_guid(creator);
+        create_object_from_guid(signer::address_of(creator), guid)
+    }
+
+    public fun create_object_from_object(creator: &signer): CreatorRef acquires Object {
```

Comment:
create_object_from_object looks exactly like create_object_from_account. What's the diff?

Reply:
where the guid is created